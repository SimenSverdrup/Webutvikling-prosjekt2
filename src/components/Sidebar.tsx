import React, { useContext } from 'react';
import { GalleryContext } from '../context/GalleryContext';
import GridFavourite from './GridFavourite';
import ListElement from './ListElement';

export default function Sidebar() {
    // han sa jeg skulle ha fuksjon inni funksjon (altså at det som er kommentert her
    // er inni en ny funksjon som endrer ting, altså js.delen)

    //state 

    //funksjoner som endrer state


    //ting som vises

    const { imgProvider, soundProvider, poemProvider }  = useContext(GalleryContext)!;
    
    //Legg til URL på hver ListeElement
    return (
        <div id="sidebar">
            <h3>Utstillingsvalg</h3>
            <h4>Bilde</h4>
            <ul>
                <ListElement text={'Rakett'} setter={imgProvider.setImgUrl} url={""}/>
                <ListElement text={'Tenkende'} setter={imgProvider.setImgUrl} url={""}/>
                <ListElement text={'Mandala'} setter={imgProvider.setImgUrl} url={""}/>
            </ul>
            <h4>Poesi</h4>
            <ul>
                <ListElement text={'Shakespare '} setter={poemProvider.setPoemUrl} url={"https://poetrydb.org/author/Shakespeare/lines.json"}/>
                <ListElement text={'Shelly'} setter={poemProvider.setPoemUrl} url={"https://poetrydb.org/title/Ozymandias/lines.json"} />
                <ListElement text={'Downson'} setter={poemProvider.setPoemUrl} url={"https://poetrydb.org/author/Ernest%20Dowson/lines.json"}/>
            </ul>
            <h4>Lyd</h4>
            <ul>
                <ListElement text={'Alt. 1'} setter={soundProvider.setSoundUrl} url={""}/>
                <ListElement text={'Alt. 2'} setter={soundProvider.setSoundUrl} url={""}/>
            </ul>

            <GridFavourite />

        </div>
    )
}