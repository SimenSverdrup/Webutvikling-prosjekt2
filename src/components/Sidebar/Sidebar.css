#sidebar {
    width: 200px;
    height: 94vh; /* Var 100% */
    background-color: #EEEEEE;
    position: absolute;
    margin: 0 auto 0 0;
  }
  
  #sidebar > h3 {
    margin-left: 5%;
  }
  
  #sidebar > h4 {
    margin-left: 15%;
    margin-top: 20%;
    margin-bottom: 2%;
  }
  
  #sidebar > ul {
    margin-left: 15%;
    cursor: pointer;
  }
  
  #sidebar > h4 {
    margin-left: 15%;
  }

  /* Veldig smal pc-skjerm/mobil hÃ¸ykant. Hadde tidligere 765 */
 @media only screen and (max-width: 683px) {
  #sidebar {
    display: grid;
    grid-template-columns: 12% 11% 11% 11% 11% 11% 11% 11% 11%;
    height: 135px;
    width: 100%;
    background-color: #EEEEEE;
    position: absolute;
    margin: 0 0 5% 0;
  }

  #sidebar > h4 {
    margin-left: 15%;
    margin-top: 5%;
    margin-bottom: 2%;
  }

  #sidebar > ul, h4 {
    margin: 0 auto 0 0;
  }

  #imgTitleList {
    grid-column-start: 1;
    grid-column-end: 4;
    grid-row-start: 1;
    grid-row-end: 2;
  }

  #poemTitleList {
    grid-column-start: 4;
    grid-column-end: 7;
    grid-row-start: 1;
    grid-row-end: 2;
  }

  #audTitleList {
    grid-column-start: 7;
    grid-column-end: 10;
    grid-row-start: 1;
    grid-row-end: 2;
  }

  #imgList {
    grid-column-start: 1;
    grid-column-end: 4;
    grid-row-start: 2;
    grid-row-end: 3;
  }

  #poemList {
    grid-column-start: 4;
    grid-column-end: 7;
    grid-row-start: 2;
    grid-row-end: 3;
  }

  #audList {
    grid-column-start: 7;
    grid-column-end: 10;
    grid-row-start: 2;
    grid-row-end: 3;
  }

  h3 {
    display: none;
  }

  h4 {
    font-size: 14px;
  }

  li {
    font-size: 13px;
  }
 }