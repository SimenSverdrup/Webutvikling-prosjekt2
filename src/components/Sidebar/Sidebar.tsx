import React, { useContext } from 'react';
import { GalleryContext } from '../../context/GalleryContext';
import Button from '../Button/Button';
import GridFavourite from '../Button/Button';
import ListElement from '../ListElement/ListElement';
import './Sidebar.css';


export default function Sidebar() {
    
    const { imgProvider, soundProvider, poemProvider }  = useContext(GalleryContext)!;
    
    return (
        <div className="sidebar"  id="sidebar">
            <h3 className="titleList" id="titleList">Utstillingsvalg</h3>
            <h4 className="imgTitleList" id="imgTitleList">Bilde</h4>
            <ul className="list" id="imgList">
                <ListElement text={'Rakett'} setter={imgProvider.setImgUrl} url={"../assets/SVG/Rocket.svg"}/>
                <ListElement text={'Tenkende'} setter={imgProvider.setImgUrl} url={"../assets/SVG/Thinking.svg"}/>
                <ListElement text={'Mandala'} setter={imgProvider.setImgUrl} url={"../assets/SVG/AnimatedMandala.svg"}/>
            </ul>
            <h4 id="poemTitleList">Poesi</h4>
            <ul className="list" id="poemList">
                <ListElement text={'Shakespeare'} setter={poemProvider.setPoemUrl} url={"https://poetrydb.org/author/Shakespeare/lines.json"}/>
                <ListElement text={'Shelley'} setter={poemProvider.setPoemUrl} url={"https://poetrydb.org/title/Ozymandias/lines.json"} />
                <ListElement text={'Downson'} setter={poemProvider.setPoemUrl} url={"https://poetrydb.org/author/Ernest%20Dowson/lines.json"}/>
            </ul>
            <h4 id="audTitleList">Lyd</h4>
            <ul className="list" id="audList">
                <ListElement text={'Pachelbel'} setter={soundProvider.setSoundUrl} url={"Canon"}/>
                <ListElement text={'Applause'} setter={soundProvider.setSoundUrl} url={"Applause"}/>
                <ListElement text={'Boo'} setter={soundProvider.setSoundUrl} url={"Boo"}/>
            </ul>
            <Button id={'saveBut'} text={'Lagre favoritt'}/>
            <Button id={'loadBut'} text={'Hent favoritt'}/>
        </div>
    )
}