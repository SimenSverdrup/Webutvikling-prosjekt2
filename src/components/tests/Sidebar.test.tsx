import React from 'react';
import renderer from 'react-test-renderer';
import Sidebar from '../Sidebar';

it('renders correctly when there are no assets', () => {
    const tree = renderer.create(<Sidebar />).toJSON();
    expect(tree).toMatchSnapshot();
});

/*
it('renders correctly when there is one single asset', () => {
    const assets = ['one'];
    const tree = renderer.create(<Listelements assets={assets} />).toJSON();
    expect(tree).toMatchSnapshot();
});



it('renders correctly when there are multiple assets', () => {
    const assets = ['one', 'two', 'three'];
    const tree = renderer.create(<Listelements assets={assets} />).toJSON();
    expect(tree).toMatchSnapshot();
});
*/
