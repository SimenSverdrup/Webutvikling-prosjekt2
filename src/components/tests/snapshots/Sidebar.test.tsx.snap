// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Container correctly 1`] = `
<div
  className="Sidebar"
>
  <button
    id="saveBut"
    onClick={[Function]}
  >
    1
  </button>
  <button
    id="loadBut"
    onClick={[Function]}
  >
    2
  </button>

</div>
`;

